tasks/service.js

```diff
@@ -1,196 +1,175 @@
+		const done = this.async();
+		const options = this.options({
+			failOnError: false,
+			stdio: 'pipe'
+		});
 
-  loopUntil = function(fnEndOk, done) {
-    var loopId;
-    return loopId = setInterval(function() {
-      if (true === fnEndOk()) {
-        clearInterval(loopId);
-        return done();
-      }
-    }, 200);
-  };
+		function killByPid(callback) {
+			if (!fs.existsSync(data.pidFile)) {
+				log.writeln("Service " + target + " - pid file not exists");
+				if (data.failOnError) {
+					return;
+				} else {
+					return callback();
+				}
+			}
+			const pid = parseInt(fs.readFileSync(data.pidFile));
+			log.writeln("Service " + target + "(pid=" + pid + ") is killing ");
+			try {
+				process.kill(pid);
+			} catch (_error) {
+				log.writeln("Service " + target + "(pid=" + pid + ") does not exists.");
+				return callback();
+			}
+			return loopUntil(function() {
+				return !existProcess(pid);
+			}, function() {
+				log.writeln("Service " + target + "(pid=" + pid + ") is killed.");
+				return callback();
+			});
+		}
+		function start(callback) {
 
-  module.exports = function(grunt) {
-    var child_process;
-    child_process = require('child_process');
-    log = grunt.log;
-    return grunt.registerMultiTask('service', 'background service', function(arg1) {
-      var data, done, killByPid, options, self, start, target;
-      if (arg1 == null) {
-        arg1 = 'start';
-      }
-      self = this;
-      target = this.target;
-      data = this.data;
-      options = this.options({
-        failOnError: false,
-        stdio: 'pipe'
-      });
-      done = this.async();
-      killByPid = function(callback) {
-        var err, pid;
-        if (!fs.existsSync(data.pidFile)) {
-          log.writeln("[Service] " + target + " - pid file not exists");
-          if (data.failOnError) {
-            return;
-          } else {
-            return callback();
-          }
-        }
-        pid = parseInt(fs.readFileSync(data.pidFile));
-        log.writeln("[Service] " + target + "(pid=" + pid + ") is killing ");
-        try {
-          process.kill(pid);
-        } catch (_error) {
-          err = _error;
-          log.writeln("[Service] " + target + "(pid=" + pid + ") does not exists.");
-          return callback();
-        }
-        return loopUntil(function() {
-          return !existProcess(pid);
-        }, function() {
-          log.writeln("[Service] " + target + "(pid=" + pid + ") is killed.");
-          return callback();
-        });
-      };
-      start = function(callback) {
-        var _closed, _spawned, args, command, pid, proc, shellCommand;
+			let buffer = "";
+			let hasStarted = false;
+			let pid;
+			let proc;
+			let command;
+			let args;
 
-		    var buffer = "";
-		    var hasStarted = false;
+			const spawned = function() { return data.blocking ? undefined : callback(); };
+			const closed = function() { return data.blocking ? callback() : undefined; };
```

Comment:
I am starting to see arrow functions as a way of declaring that a function cannot change the context in which it is run.  So it's actually a stricter form of declaration than `function` and should probably be preferred in this case.